File: task2.py
import africastalking
from flask import Flask, request
import re

#Create your credentials
username = "sandbox"
apikey = "e02f74dbffdcdfc13d356890e7dfff3769f5e8c0b2dd92b2d98b81989dabb5d3"

#Initialize the SDK
africastalking.initialize(username, apikey)

#Get the SMS service
sms = africastalking.SMS

app = Flask(_name_)

response = ""


def sendconfirmation(recipients, message, sender):
    #Send the SMS
    try:
        #Once this is done, that's it! We'll handle the rest
        response = sms.send(message, recipients, sender)
        print(response)
    except Exception as e:
        print(f"message wasn't sent because {e}")



@app.route('/', methods=['POST', 'GET'])
def ussd_callback():

    global response
    session_id = request.values.get("sessionId", None)
    service_code = request.values.get("serviceCode", None)
    phone_number = request.values.get("phoneNumber", None)
    text = request.values.get("text", "default")
    #used regex to check that user name was either a single word or two words separated by a space
    usernamePattern1 = re.compile("\w+\s\w+")
    usernamePattern2 = re.compile("\w+")

    # asks user to enter their name
    if text == '':
        response  = "CON Enter your user name\n"
    # after entering their name ask the user to enter their email address
    elif usernamePattern1.match(text) or usernamePattern2.match(text):
        response = "CON Enter your email address\n"
    else:
        response  = "END You will receive an sms confirmation shortly\n"
        sendconfirmation([phone_number],"You have successfully registered",service_code)

  return response

File: Task2.php
<?php
//composer dependencies
require 'vendor/autoload.php';
use AfricasTalking\SDK\AfricasTalking;

// Reads the variables sent via POST
$sessionId   = $_POST["sessionId"];
$serviceCode = $_POST["serviceCode"];
$phoneNumber = $_POST["phoneNumber"];
$text        = $_POST["text"];



//This is the first menu 
if ( $text == "" ) {
    $response  = "CON Welcome Client, Please Register \n";
    $response .= "1. Enter your Username \n";    
}



// Menu after inputing username
else if ($text == $text) {
    $response  = "CON Email Address \n";
    $response .= "1. Enter your email address\n";
}


//Message after user finishes registration
else if ($text == preg_replace('/^[a-zA-Z0-9]+$/', $str)){
    $response = "Successfully Registered\n";
}

// Print the response onto the page so that our gateway can read it
header('Content-type: text/plain');
echo $response;


////SMS Section

$username   = "sandbox";
$apiKey     = "0c874ea406208f3ae1f2c1336f09db60772a139dd57b5ea30c72662e0e5c71b4";

// Initialize the SDK
$AT         = new AfricasTalking($username, $apiKey);

// Get the SMS service
$sms        = $AT->sms();

// Recepients mobile number
$recipients = $phoneNumber;

// message
$message    = "You have registered successfully"; 

// Set your shortCode or senderId
$from       = "5ukuma";

try {
    $result = $sms->send([
        'to'      => $recipients,
        'message' => $message,
        'from'    => $from
    ]);

    print_r($result);
} catch (Exception $e) {
    echo "Error: ".$e->getMessage();
}


?>

